### ISSUE TITLE
Add section on sub-field formatting

### ISSUE THREAD

**Issue by bdarcus** [citation-style-language/documentation#104]
Evolving out of citation-style-language/schema#278, we should add a section on this, to at least establish the baseline sub-field formatting features of:

- italic
- bold
- verbatim/code
- quotes
- preserve case

**Comment by bwiernik** [comment]
We need preserve case for at least two cases:
1. Preventing uppercasing of species names in Title Case. 
2. Preventing upper casing of things like software titles at the beginning of titles. 

Strikethrough is occasionally used in titles, usually for light comic effect.

**Comment by bdarcus** [comment]
> I think this can apply to any name or standard variable.

Really? Maybe a couple/few examples, with at least one name?

**Comment by bwiernik** [comment]
e. e. cummings should not be uppercased ever

**Comment by bdarcus** [comment]
Great example. Also "bell hooks".

>
But where would that be exposed as a problem?

Don't we assume no messing with case in names; that the data represents the
proper casing?

I see it is a problem if the name is in a title, because while a processor
can know how to title case sentence case, it will fail on names that are
all lower?

**Comment by bwiernik** [comment]
We permit `@text-case` on `cs:name-part`.

**Comment by bwiernik** [comment]
In terms of common field markup, we should also consider underlining.

**Comment by bdarcus** [comment]
How would that relate to italic/emphasis?

**Comment by bwiernik** [comment]
Separate (e.g., `<i></i>` versus `<u></u>`). The specific use case I am encountering at the moment is that my university wants graduate student names in bold and my name underlined for our CV formatting.

**Comment by bdarcus** [comment]
This is tied to a larger discussion of how we will deal with mapping input
subfield formatting to output.

Generally speaking, over the long history of typography, italics and
underlines have both been used to indicate some kind of emphasis; to set it
off from the base text formatting.

In bibliographic data, that data is often representing titles, names, etc.

What I'm wondering about is this:

Underlining is an archaic practice completely unnecessary, and
arguably exceedingly tacky, in the 21st century.

So what if we have a style that takes that position, and insists underline
not be used in output, and instead some other convention (bold or italic)?
For this example, let's say italic.

What happens if we have an input field with both underline and italic, and
the output for the field as a whole is italic?

Not sure if I'm explaining this well; do you get what I'm saying?

**Comment by bwiernik** [comment]
I understand the history and context. What I am saying is that from the perspective of CSL, underline and italics should be treated as completely independent of each other. Italics is italics. Underline is underline. They have nothing otherwise to do with each other. Just like how we call underlining "text-decoration" in CSL styles that that has no interaction with "font-face='italic'".

If the style field is italic and a word is marked up to be underlined, that word would be italic and underlined. If it were marked up to be italic and underlined, that word would be non-italic and underlined (italic flip-flopping). We could potentially implement general flip-flopping for bold/italic/underline, but I don't see much demand for that.

I certainly agree that underlining is generally archaic, but underlinining is (1) one of the few major text decoration features we aren't supporting and (2) especially for one-off field markup like highlighting a specific name or group of names in a reference list, is a format that generally doesn't conflict with other styling (e.g., most styles use italics, so it doesn't really stand out if italics is also used for emphasis).

**Comment by bdarcus** [comment]
> If the style field is italic and a word is marked up to be underlined, that word would be italic and underlined. If it were marked up to be italic and underlined, that word would be non-italic and underlined (italic flip-flopping).

And what if a style doesn't allow underline? Is that out of scope for us, at least for now?

**Comment by bwiernik** [comment]
Yeah, I don't think we intervene between styles and in-field markup. That's similar to other data entry concerns.
